test/LanguageExpressionsTest.cpp

```diff
@@ -0,0 +1,274 @@
+    context._endIndex = table.size();
+
+    // Define the mapping from variable names to column indices.
+    using V = Variable;
+    varToColMap[V{strVar[0]}] = makeAlwaysDefinedColumn(0);
+    varToColMap[V{strVar[1]}] = makeAlwaysDefinedColumn(1);
+  }
+};
+
+// ____________________________________________________________________________
+auto litOrIri = [](const std::string& literal) {
+  return LiteralOrIri::fromStringRepresentation(
+      absl::StrCat("\""sv, literal, "\""sv));
+};
+
+// ____________________________________________________________________________
+auto assertLangTagValueGetter =
+    [](const std::vector<Id>& input, const std::vector<strOpt>& expected,
+       LanguageTagGetter& langTagValueGetter, TestContext& testContext) {
+      EXPECT_EQ(input.size(), expected.size());
+      auto ctx = &testContext.context;
+      for (size_t i = 0; i < input.size(); i++) {
+        ASSERT_EQ(expected[i], langTagValueGetter(input[i], ctx));
+      }
+    };
+
+// ____________________________________________________________________________
+auto getLangExpression =
+    [](const std::string& variable) -> SparqlExpression::Ptr {
+  // initialize with a child expression which has to be a VariableExpression!
+  return std::make_unique<LangExpression>(
+      std::make_unique<VariableExpression>(Variable{variable}));
+};
+
+// ____________________________________________________________________________
+auto getLangMatchesExpression =
+    [](const std::string& variable,
+       const std::string& langRange) -> SparqlExpression::Ptr {
+  return makeLangMatchesExpression(
+      getLangExpression(variable),
+      std::make_unique<StringLiteralExpression>(lit(langRange)));
+};
+
+// ____________________________________________________________________________
+template <auto GetExpr, typename T>
+auto testLanguageExpressions = [](const std::vector<T>& expected,
+                                  const std::string& variable,
+                                  const auto&... args) {
+  TestContext context;
+  ASSERT_TRUE(context.isValidVariableStr(variable));
+  SparqlExpression::Ptr expr = GetExpr(variable, args...);
+  auto resultAsVariant = expr->evaluate(&context.context);
+  const auto& result = std::get<VectorWithMemoryLimit<T>>(resultAsVariant);
+  EXPECT_THAT(result, ::testing::ElementsAreArray(expected));
+};
+
+// ____________________________________________________________________________
+TEST(LanguageTagGetter, testLanguageTagValueGetterWithoutVocabId) {
+  TestContext testContext{};
+  LanguageTagGetter langTagGetter{};
+  Id dateId1 = Id::makeFromDate(
+         DateOrLargeYear::parseXsdDatetime("1900-12-13T03:12:00.33Z")),
+     dateId2 = Id::makeFromDate(DateOrLargeYear::parseXsdDate("2025-01-01")),
+     boolId1 = Id::makeFromBool(false), boolId2 = Id::makeFromBool(true),
+     intId = Id::makeFromInt(323), doubleId = Id::makeFromDouble(234.23);
+  Id maxId = Id::max(), U = Id::makeUndefined();
```

Comment:
some of those already have short functions (the `IntId` or `DoubleId`, and if something like this is missing for `bool`, and dates (parse from string, then make Id), then consider adding them.